datadog_checks_downloader/tests/test_downloader.py

```diff
@@ -37,39 +64,289 @@
+@mock.patch("time.time", mock.MagicMock(return_value=_LOCAL_TESTS_DATA_TIMESTAMP))
+def test_local_wheels_signer_signature_leaf_error(distribution_name, distribution_version):
+    """Test failure in verifying wheels-signer signature.
+
+    The wheel-signer-{a-z} metadata has to have wrong signature.
+    """
+    test_data = "{}-{}-signature-wheels-signer-a".format(distribution_name, distribution_version)
+    with local_http_server(test_data) as http_url:
+        argv = [
+            distribution_name,
+            "--version",
+            distribution_version,
+            "--repository",
+            http_url,
+        ]
+
+        with pytest.raises(NoWorkingMirrorError) as exc:
+            _do_run_downloader(argv)
+
+    assert "BadSignatureError('wheels-signer-" in str(exc)
+
+
 def delete_files(files):
     for f in files:
         os.remove(f)
 
 
-def cleanup():
-    REPO_DIR = 'datadog_checks/downloader/data/repo/'
+def _cleanup():
+    REPO_DIR = "datadog_checks/downloader/data/repo/"
 
-    METADATA_DIR = os.path.join(REPO_DIR, 'metadata')
-    TARGETS_DIR = os.path.join(REPO_DIR, 'targets')
-    IN_TOTO_METADATA_DIR = os.path.join(TARGETS_DIR, 'in-toto-metadata')
-    IN_TOTO_PUBKEYS_DIR = os.path.join(TARGETS_DIR, 'in-toto-pubkeys')
-    SIMPLE_DIR = os.path.join(TARGETS_DIR, 'simple')
+    METADATA_DIR = os.path.join(REPO_DIR, "metadata")
+    TARGETS_DIR = os.path.join(REPO_DIR, "targets")
+    IN_TOTO_METADATA_DIR = os.path.join(TARGETS_DIR, "in-toto-metadata")
+    IN_TOTO_PUBKEYS_DIR = os.path.join(TARGETS_DIR, "in-toto-pubkeys")
+    SIMPLE_DIR = os.path.join(TARGETS_DIR, "simple")
 
     # First, nuke all known targets. but not the directory for targets itself.
-    shutil.rmtree(IN_TOTO_METADATA_DIR)
-    shutil.rmtree(IN_TOTO_PUBKEYS_DIR)
-    shutil.rmtree(SIMPLE_DIR)
+    shutil.rmtree(IN_TOTO_METADATA_DIR, ignore_errors=True)
+    shutil.rmtree(IN_TOTO_PUBKEYS_DIR, ignore_errors=True)
+    shutil.rmtree(SIMPLE_DIR, ignore_errors=True)
 
     # Then, nuke all previous TUF metadata.
-    previous_jsons = os.path.join(METADATA_DIR, 'previous/*.json')
+    previous_jsons = os.path.join(METADATA_DIR, "previous/*.json")
     previous_jsons = glob.glob(previous_jsons)
     delete_files(previous_jsons)
 
     # Finally, nuke ALL current TUF metadata EXCEPT the unversioned root metadata.
-    current_jsons = os.path.join(METADATA_DIR, 'current/*.json')
+    current_jsons = os.path.join(METADATA_DIR, "current/*.json")
     current_jsons = glob.glob(current_jsons)
-    current_jsons = [c for c in current_jsons if os.path.basename(c) != 'root.json']
+    current_jsons = [c for c in current_jsons if os.path.basename(c) != "root.json"]
     delete_files(current_jsons)
 
 
 @retry(wait=wait_exponential(min=2, max=60), stop=stop_after_attempt(10))
-def download(package, version=None, root_layout_type='core'):
+def _do_download(package, version=None, root_layout_type="core"):
```

Comment:
I am a python newbie, but why change `download()` to `_do_download()`?

Reply:
The function name would shadow the one imported:

```python
from datadog_checks.downloader.cli import download
```